src/mgmt/srv.c

```diff
@@ -59,23 +60,104 @@ static struct daos_rpc_handler mgmt_handlers[] = {
 
 #undef X
 
+static void
+process_killrank_request(Drpc__Call *drpc_req, Proto__DaosResponse *daos_resp)
+{
+	Proto__DaosRank	*pb_rank = NULL;
+
+	proto__daos_response__init(daos_resp);
+
+	/* Unpack the daos request from the drpc call body */
+	pb_rank = proto__daos_rank__unpack(
+		NULL, drpc_req->body.len, drpc_req->body.data);
+
+	if (pb_rank == NULL) {
+		daos_resp->status = PROTO__DAOS_REQUEST_STATUS__ERR_UNKNOWN;
+		D_ERROR("Failed to extract rank from request\n");
+
+		return;
+	}
+
+	/* response status is populated with SUCCESS on init */
+	D_DEBUG(DB_MGMT, "Received request to kill rank (%u) on pool (%s)\n",
+		pb_rank->rank, pb_rank->pool_uuid);
+
+	/* TODO: do something with request and populate daos response status */
+
+	proto__daos_rank__free_unpacked(pb_rank, NULL);
+}
+
+static void
+process_drpc_request(Drpc__Call *drpc_req, Drpc__Response *drpc_resp)
+{
+	Proto__DaosResponse	*daos_resp = NULL;
+	uint8_t			*body = NULL;
+	size_t			len = 0;
+
+	D_ALLOC_PTR(daos_resp);
+	if (daos_resp == NULL) {
+		drpc_resp->status = DRPC__STATUS__FAILURE;
+		D_ERROR("Failed to allocate daos response ref\n");
+
+		return;
+	}
+
+	switch (drpc_req->method) {
+	case DRPC_METHOD_MGMT_KILL_RANK:
+		/**
+		 * Process drpc request and populate daos response,
+		 * command errors should be indicated inside daos response.
+		 */
+		process_killrank_request(drpc_req, daos_resp);
+
+		len = proto__daos_response__get_packed_size(daos_resp);
+		D_ALLOC(body, len);
+		if (body == NULL) {
+			drpc_resp->status = DRPC__STATUS__FAILURE;
```

Comment:
Not blocking for this PR but do we want to expand the DRPC error space to handle inability to allocate or marshal responses?

Reply:
yes